itkwidgets/widget_viewer.py

```diff
@@ -39,6 +45,10 @@ def view(image):
     if isinstance(image, np.ndarray):
         image_from_array = itk.GetImageViewFromArray(image)
         viewer.image = image_from_array
+    elif have_dask and isinstance(image, dask.array.core.Array):
+        array = np.array(image)
+        image_from_array = itk.GetImageViewFromArray(array)
+        viewer.image = image_from_array
```

Comment:
Two questions here:

1.  Is the passed in image the full dask array, or is this after something has sliced out the appropriate pieces from it
2.  Are there opportunities to do some of the preprocessing with dask ahead of time?
3.  I wonder if code like this could avoid the explicit type check, and instead rely on duck typing.  For example I wonder if it could be merged with the ndarray type check above with something like the following:

```python
def is_arraylike(x):
    return hasattr(x, 'shape') and hasattr(x, 'dtype') and hasattr(x, '__array__'):

if is_arraylike(x):
    x = np.asarray(x)
    image_from_array = itk.GetImageViewFromArray(image)
    viewer.image = image_from_array
```

This might both avoid the explicit dependence on dask.array and also open you up to other array-like objects beyond just dask.arrays.  (for more thoughts, see http://matthewrocklin.com/blog/work/2018/05/27/beyond-numpy)

Reply:
Great thoughts @mrocklin !

> Is the passed in image the full dask array, or is this after something has sliced out the appropriate pieces from it?

Correct, it is the full dask array, but Task 2, Task 3 of #44 would enable grabbing slices.

> Are there opportunities to do some of the preprocessing with dask ahead of time?

Yes, we could use dask for processing or daa or downsampling, possibly with `coarsen`, as recommended.

> I wonder if code like this could avoid the explicit type check, and instead rely on duck typing. 

Great idea. I'll take a look and test it out.

Reply:
Love the ideas in the blog. That should be a worthwhile efforts.

I tested your suggestion with `is_arraylike` and it works well. I made a minor tweak -- added `ndim`, which is needed in this case.

Thanks again for the review!